django/db/models/fields/related_descriptors.py

```diff
@@ -172,23 +171,26 @@ def __get__(self, instance, cls=None):
         if instance is None:
             return self
 
-        # The related instance is loaded from the database and then cached in
-        # the attribute defined in self.cache_name. It can also be pre-cached
+        # The related instance is loaded from the database and then cached
+        # by the field on the model instance state. It can also be pre-cached
         # by the reverse accessor (ReverseOneToOneDescriptor).
         try:
-            rel_obj = getattr(instance, self.cache_name)
-        except AttributeError:
+            rel_obj = self.field.get_cached_value(instance)
```

Comment:
This is one of the cases I mentioned above, if we replace this with a default of None then we risk running a computation (the block after keyerror) when the cache for rel_obj was explicitly set to None.